ludwig/schema/hyperopt/search_algorithm.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import field
+from typing import Dict
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.schema import utils as schema_utils
+
+
+@dataclass
+class SearchAlgorithmConfig(schema_utils.BaseMarshmallowConfig):
+    """Basic search algorithm settings."""
+
+    type: str = schema_utils.StringOptions(options=["hyperopt"], default="hyperopt", allow_none=False)
```

Comment:
Same here - there's definitely more options that we theoretically support in Ludwig: https://docs.ray.io/en/releases-1.12.1/tune/api_docs/suggestion.html#search-algorithms-tune-suggest, so just allowing hyperopt as the type might be very restrictive for Ludwig users if we merge this in.

Reply:
will add in

Reply:
done